### PR TITLE
overlay lib: Secure the creation of a disk image when size is < 130

### PR DESCRIPTION
**Pull request by pirat89**
In case the filesystem for which the disk img is going to be created has very small amount of free space (under 130 MiBs), it cannot be formatted by XFS with current params.

This could be hit in several cases:
 * the system partition/volume - in this case, most likely an issue will be hit anyway later by DNF speaking about small amount of free space if a content is installed inside by RPMs as such a small amount of free space is really not expected to see at all
 * it's a data mount point (e.g. iso) or a filesystem type that should by part of the OVERLAY_DO_NOT_MOUNT set, so enlarging the value to 130 MiBs should not affect anything negatively at all
 * in case of /boot, the problem with the free space is covered already in a different actor prior we try to create any disk img, so we are safe here

Based on arguments above, I am considering setting the 130 MiBs as minimal value safe for in-place upgrades. Also it will allow to skip possible problems with specific file systems (like tmpfs, ...) in case we are still missing some in the OVERLAY_DO_NOT_MOUNT - and kind of read only storage (such as iso9660, etc..).

JIRA: OAMG-9463

## Orig error

Also updating the error msg when a FS cannot be created as originally it contained just:
```
2023-07-31 12:31:27.48  DEBUG    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has started: ['/bin/dd', 'if=/dev/zero', 'of=/var/lib/leapp/scratch/diskimages/root_home', 'bs=1M', 'count=0', 'seek=100']
0+0 records in
0+0 records out
0 bytes (0 B) copied, 9.9531e-05 s, 0.0 kB/s
2023-07-31 12:31:27.57  DEBUG    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has finished: ['/bin/dd', 'if=/dev/zero', 'of=/var/lib/leapp/scratch/diskimages/root_home', 'bs=1M', 'count=0', 'seek=100']
2023-07-31 12:31:27.58  DEBUG    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: Creating XFS filesystem in disk image at /var/lib/leapp/scratch/diskimages/root_home
2023-07-31 12:31:27.59  DEBUG    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has started: ['/sbin/mkfs.xfs', '-l', 'size=32m', '-f', '/var/lib/leapp/scratch/diskimages/root_home']
internal log size 8192 too large, must fit in allocation group
Usage: mkfs.xfs
/* blocksize */		[-b log=n|size=num]
/* metadata */		[-m crc=0|1,finobt=0|1,uuid=xxx]
/* data subvol */	[-d agcount=n,agsize=n,file,name=xxx,size=num,
			    (sunit=value,swidth=value|su=num,sw=num|noalign),
			    sectlog=n|sectsize=num
/* force overwrite */	[-f]
/* inode size */	[-i log=n|perblock=n|size=num,maxpct=n,attr=0|1|2,
			    projid32bit=0|1]
/* no discard */	[-K]
/* log subvol */	[-l agnum=n,internal,size=num,logdev=xxx,version=n
			    sunit=value|su=num,sectlog=n|sectsize=num,
			    lazy-count=0|1]
/* label */		[-L label (maximum 12 characters)]
/* naming */		[-n log=n|size=num,version=2|ci,ftype=0|1]
/* no-op info only */	[-N]
/* prototype file */	[-p fname]
/* quiet */		[-q]
/* realtime subvol */	[-r extsize=num,size=num,rtdev=xxx]
/* sectorsize */	[-s log=n|size=num]
/* version */		[-V]
			devicename
<devicename> is required unless -d name=xxx is given.
<num> is xxx (bytes), xxxs (sectors), xxxb (fs blocks), xxxk (xxx KiB),
      xxxm (xxx MiB), xxxg (xxx GiB), xxxt (xxx TiB) or xxxp (xxx PiB).
<value> is xxx (512 byte blocks).
2023-07-31 12:31:27.88  DEBUG    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: Command ['/sbin/mkfs.xfs', '-l', 'size=32m', '-f', '/var/lib/leapp/scratch/diskimages/root_home'] failed with exit code 1.
2023-07-31 12:31:27.90  DEBUG    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has finished: ['/sbin/mkfs.xfs', '-l', 'size=32m', '-f', '/var/lib/leapp/scratch/diskimages/root_home']
2023-07-31 12:31:27.91  ERROR    PID: 15851 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: Failed to create XFS filesystem /var/lib/leapp/scratch/diskimages/root_home
Traceback (most recent call last):
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/overlaygen.py", line 386, in _format_disk_image_xfs
    utils.call_with_oserror_handled(cmd=cmd)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/utils.py", line 119, in call_with_oserror_handled
    run(cmd)
  File "/usr/lib/python2.7/site-packages/leapp/libraries/stdlib/__init__.py", line 188, in run
    result=result
CalledProcessError: Command ['/sbin/mkfs.xfs', '-l', 'size=32m', '-f', '/var/lib/leapp/scratch/diskimages/root_home'] failed with exit code 1.
....
============================================================
                           ERRORS                           
============================================================

2023-07-31 12:31:27.096968 [ERROR] Actor: target_userspace_creator
Message: Command ['/sbin/mkfs.xfs', '-l', 'size=32m', '-f', '/var/lib/leapp/scratch/diskimages/root_home'] failed with exit code 1.

============================================================
                       END OF ERRORS                        
============================================================
```

which has not been saying so much. It should be now better in any way when a FS cannot be created for whatever reason

### CODE REVIEW

**Comment by pirat89** [comment]
@oamg/developers  another aproach would be just to keep the error message. but I started to think about cases that we are still missing and that there could be actually some mountpoints that cannot be enlarged (like iso...not sure why someone would put that in fstab, but...) or mounpoints that we should actually ignore. wdyt about that? In case we do not want to set a minimum limit, we can just update the error msg in a way we inform user what partition/volume does not contain enough free space.

Need to still keep in mind that this whole problem is about corner cases.

File: repos/system_upgrade/common/libraries/overlaygen.py [hunk 1]
```diff
@@ -405,15 +411,42 @@ def _create_mount_disk_image(disk_images_directory, path, disk_size):
...
+        #    read only (e.g. ISO), or it's an FS type that should be covered by
+        #    OVERLAY_DO_NOT_MOUNT
+        #  * most common case important for us here could be /boot, but that's
+        #    covered already in different actors/checks, so it should not be
+        #    problem either
+        # NOTE(pstodulk): In case the formatting params are modified,
+        # the minimal required size could be different
+        api.current_logger().warning(
+            'The apparent size for the disk image representing {path}'
+            ' is too small ({} MiBs) for a formatting. Setting 130 MiBs instead.'
```
Comment (MichalHe) [inline repos/system_upgrade/common/libraries/overlaygen.py:447]: This warning does not have its placeholders formatted (a missing `.format` perhaps)

File: repos/system_upgrade/common/libraries/overlaygen.py [hunk 2]
```diff
@@ -388,7 +391,10 @@ def _format_disk_image_xfs(diskimage_path):
         # FIXME(pstodulk): taken from original, but %s seems to me invalid here
         api.current_logger().error('Failed to create XFS filesystem %s', diskimage_path, exc_info=True)
         raise StopActorExecutionError(
-            message=str(e)
+            message='Cannot create XFS filesystem in{}'.format(diskimage_path),
```
Comment (MichalHe) [inline repos/system_upgrade/common/libraries/overlaygen.py:394]:
```suggestion
            message='Cannot create XFS filesystem in {}'.format(diskimage_path),
```

File: repos/system_upgrade/common/libraries/overlaygen.py [hunk 3]
```diff
@@ -367,9 +367,12 @@ def _format_disk_image_ext4(diskimage_path):
         utils.call_with_oserror_handled(cmd=cmd)
     except CalledProcessError as e:
         # FIXME(pstodulk): taken from original, but %s seems to me invalid here
-        api.current_logger().error('Failed to create ext4 filesystem %s', diskimage_path, exc_info=True)
+        api.current_logger().error('Failed to create ext4 filesystem in %s', diskimage_path, exc_info=True)
         raise StopActorExecutionError(
-            message=str(e)
+            message='Cannot create Ext4 filesystem in{}'.format(diskimage_path),
```
Comment (MichalHe) [inline repos/system_upgrade/common/libraries/overlaygen.py:372]:
```suggestion
            message='Cannot create Ext4 filesystem in {}'.format(diskimage_path),
```

**Review by MichalHe — CHANGES_REQUESTED** [review]
I've tested the patch with a `256mb` partition that contains `200mb` file, and the patch works as expected - preupgrade finished without problems, whereas without this patch the preupgrade fails.

Logs produced by this patch:
```
2023-08-16 14:45:48.454 WARNING  PID: 15114 leapp.workflow.TargetTransactionCheck.dnf_transaction_check: The apparent size for the disk image representing {path} is too small ({} MiBs) for a formatting. Setting 130 MiBs instead.
2023-08-16 14:45:48.455 DEBUG    PID: 15114 leapp.workflow.TargetTransactionCheck.dnf_transaction_check: Attempting to create disk image at /var/lib/leapp/scratch/diskimages/root_mnt
2023-08-16 14:45:48.456 DEBUG    PID: 15114 leapp.workflow.TargetTransactionCheck.dnf_transaction_check: External command has started: ['/bin/dd', 'if=/dev/zero', 'of=/var/lib/leapp/scratch/diskimages/root_mnt', 'bs=1M', 'count=0', 'seek=130']
```
@pirat89 please fix the minor issues found and I will approve this no problemo.

**Review by MichalHe — APPROVED** [review]

**Comment by pirat89** [comment]
Tests passed. Merging \o

**Pull request merged**